tabpy-server/tabpy_server/handlers/base_handler.py

```diff
@@ -1,30 +1,143 @@
 import base64
 import binascii
 import concurrent
-import tornado.web
 import json
 import logging
-
+import tornado.web
 from tabpy_server.app.SettingsParameters import SettingsParameters
 from tabpy_server.handlers.util import hash_password
+import uuid
+
 
-logger = logging.getLogger(__name__)
 STAGING_THREAD = concurrent.futures.ThreadPoolExecutor(max_workers=3)
 
 
-class BaseHandler(tornado.web.RequestHandler):
-    KEYS_TO_SANITIZE = ("api key", "api_key", "admin key", "admin_key")
+class ContextLoggerWrapper(object):
+    '''
+    This class appends request context to logged messages.
+    '''
+    @staticmethod
+    def _generate_call_id():
+        return str(uuid.uuid4())
+
+    def __init__(self, request: tornado.httputil.HTTPServerRequest):
+        self.call_id = self._generate_call_id()
+        self.set_request(request)
+
+        self.tabpy_username = None
+        self.log_request_context = False
+        self.request_context_logged = False
+
+    def set_request(self, request: tornado.httputil.HTTPServerRequest):
+        '''
+        Set HTTP(S) request for logger. Headers will be used to
+        append request data as client information, Tableau user name, etc.
+        '''
+        self.remote_ip = request.remote_ip
+        self.method = request.method
+        self.url = request.full_url()
+
+        if 'TabPy-Client' in request.headers:
```

Comment:
More idiomatic to write this as:

`self.client = request.headers.get('TabPy-Client')`